sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1531,6 +1531,253 @@ def _df_getitem_unicode_idx_impl(self, idx):
     ty_checker.raise_exc(idx, expected_types, 'idx')
 
 
+def df_getitem_tuple_iat_codegen(self, row, col):
+    """
+    Example of generated implementation:
+        def _df_getitem_tuple_iat_impl(self, idx):
+            data = get_dataframe_data(self._dataframe, 1)
+            res_data = pandas.Series(data)
+            return res_data.iat[idx[0]]
+    """
+    func_lines = ['def _df_getitem_tuple_iat_impl(self, idx):']
+    if self.columns:
+        func_lines += [
+            f"  data = get_dataframe_data(self._dataframe, {col})",
+            f"  res_data = pandas.Series(data)",
+            f"  return res_data.iat[idx[0]]",
+        ]
+
+    func_text = '\n'.join(func_lines)
+
+    global_vars = {'pandas': pandas, 'numpy': numpy,
+                   'get_dataframe_data': get_dataframe_data}
+
+    return func_text, global_vars
+
+
+def gen_df_getitem_iat_impl(self, row, col):
+    df_func_name = f'_df_getitem_tuple_iat_impl'
+    func_text, global_vars = df_getitem_tuple_iat_codegen(self, row, col)
+    loc_vars = {}
+    exec(func_text, global_vars, loc_vars)
+    _reduce_impl = loc_vars[df_func_name]
+
+    return _reduce_impl
+
+
+def df_getitem_int_iloc_codegen(self, idx):
+    """
+    Example of generated implementation:
+        def _df_getitem_int_iloc_impl(self, idx):
+            data_A = pandas.Series(get_dataframe_data(self._dataframe, 0))
+            result_A = data_A.iat[idx]
+            data_B = pandas.Series(get_dataframe_data(self._dataframe, 1))
+            result_B = data_B.iat[idx]
+            data_C = pandas.Series(get_dataframe_data(self._dataframe, 2))
+            result_C = data_C.iat[idx]
+            return pandas.Series(data=[result_A, result_B, result_C], index=['A', 'B', 'C'],
+                name=str(self._dataframe._index[idx]))
+    """
+    func_lines = ['def _df_getitem_int_iloc_impl(self, idx):']
+    results = []
+    if self.columns:
+        index = []
+        name = df_name_codegen_iloc(self)
+        for i, c in enumerate(self.columns):
+            result_c = f"result_{c}"
```

Comment:
Please, use index instead column name? Use of column names is not safely, these bugs are hard to spot lately! 
```suggestion
            result_c = f"result_{i}"
```